// @storybook/addon-docs/dist/cjs/frameworks/svelte/prepareForInline.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.prepareForInline = void 0;

var _react = _interopRequireDefault(require("react"));

var _HOC = _interopRequireDefault(require("./HOC.svelte"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// @ts-ignore
var prepareForInline = function prepareForInline(storyFn, context) {
  var el = _react.default.useRef(null);

  _react.default.useEffect(function () {
    var cancelled = false;
    var applyLoaders = context.applyLoaders,
        unboundStoryFn = context.unboundStoryFn;
    var cpn;
    applyLoaders().then(function (storyContext) {
      if (!cancelled) {
        cpn = new _HOC.default({
          target: el.current,
          props: {
            storyContext: storyContext,
            unboundStoryFn: unboundStoryFn
          }
        });
      }
    });
    return function () {
      cancelled = true;

      if (cpn) {
        cpn.$destroy();
      }
    };
  });

  return /*#__PURE__*/_react.default.createElement('div', {
    ref: el
  });
};

exports.prepareForInline = prepareForInline;
